mtags/src/main/scala-2/scala/meta/internal/pc/PcInlayHintsProvider.scala

```diff
@@ -255,8 +267,10 @@ final class PcInlayHintsProvider(
     private def isCompilerGeneratedSymbol(sym: Symbol) =
       sym.decodedName.matches("x\\$\\d+")
 
-    private def isNotInUnapply(vd: ValDef) =
-      !vd.rhs.pos.isRange || vd.rhs.pos.start > vd.namePosition.end
+    private def isValidUnapply(vd: ValDef) =
+      (!vd.rhs.pos.isDefined && params.hintsInPatternMatch()) ||
```

Comment:
Could you separate it into different methods? It's a bit unclear what this does check